chainntnfs/btcdnotify/btcd.go

```diff
@@ -447,59 +450,127 @@ out:
+		if !ok || jsonErr.Code != btcjson.ErrRPCNoTxInfo {
+			return nil, fmt.Errorf("unable to query for txid "+
+				"%v: %v", txid, err)
 		}
-		return nil, fmt.Errorf("unable to query for txid(%v): %v", txid, err)
+	}
+
+	// Make sure we actually retrieved a transaction that is included in a
+	// block. Without this, we won't be able to retrieve its confirmation
+	// details.
+	if tx == nil || tx.BlockHash == "" {
+		return nil, nil
 	}
 
 	// As we need to fully populate the returned TxConfirmation struct,
 	// grab the block in which the transaction was confirmed so we can
 	// locate its exact index within the block.
 	blockHash, err := chainhash.NewHashFromStr(tx.BlockHash)
 	if err != nil {
-		return nil, fmt.Errorf("unable to get block hash %v for historical "+
-			"dispatch: %v", tx.BlockHash, err)
+		return nil, fmt.Errorf("unable to get block hash %v for "+
+			"historical dispatch: %v", tx.BlockHash, err)
 	}
+
 	block, err := b.chainConn.GetBlockVerbose(blockHash)
 	if err != nil {
-		return nil, fmt.Errorf("unable to get block hash: %v", err)
+		return nil, fmt.Errorf("unable to get block with hash %v for "+
+			"historical dispatch: %v", blockHash, err)
 	}
 
-	// If the block obtained, locate the transaction's index within the
+	// If the block was obtained, locate the transaction's index within the
 	// block so we can give the subscriber full confirmation details.
-	txIndex := -1
 	targetTxidStr := txid.String()
-	for i, txHash := range block.Tx {
+	for txIndex, txHash := range block.Tx {
 		if txHash == targetTxidStr {
-			txIndex = i
-			break
+			return &chainntnfs.TxConfirmation{
+				BlockHash:   blockHash,
+				BlockHeight: uint32(block.Height),
+				TxIndex:     uint32(txIndex),
+			}, nil
 		}
 	}
 
-	if txIndex == -1 {
-		return nil, fmt.Errorf("unable to locate tx %v in block %v",
-			txid, blockHash)
-	}
+	// We return an error because we should have found the transaction
+	// within the block, but didn't.
+	return nil, fmt.Errorf("unable to locate tx %v in block %v", txid,
+		blockHash)
+}
+
+// confDetailsManually looks up whether a transaction is already included in a
+// block in the active chain by scanning the chain's blocks, starting from the
+// earliest height the transaction could have been included in, to the current
+// height in the chain. If it is found, the confirmation details are returned.
+// Otherwise, an error is not returned.
+func (b *BtcdNotifier) confDetailsManually(txid *chainhash.Hash,
+	heightHint, currentHeight uint32) (*chainntnfs.TxConfirmation, error) {
+
+	targetTxidStr := txid.String()
+
+	// Begin scanning blocks at every height to determine where the
+	// transaction was included in.
+	for height := heightHint; height <= currentHeight; height++ {
+		blockHash, err := b.chainConn.GetBlockHash(int64(height))
+		if err != nil {
+			return nil, fmt.Errorf("unable to get hash from block "+
+				"with height %d", height)
+		}
+
+		block, err := b.chainConn.GetBlockVerbose(blockHash)
```

Comment:
Worth adding a `TODO` here that we can actually just fetch the neutrino filters here. Will save us a bit of processing as we can quickly ascertain if a transaction _maybe_ includes a block or not.

Reply:
Done.